### PR TITLE
Add appendix explaining background of feature detection

### PR DESCRIPTION
**Pull request by Acconut**

### CODE REVIEW

**Review by guoye-zhang — APPROVED** [review]

File: draft-tus-resumable-uploads-protocol.md [hunk 1]
```diff
@@ -368,3 +368,29 @@ Specification: This document
...
 
 TODO acknowledge.
+
+# Appendix
+
+## Feature Detection
+
+This specification includes a section about feature detection (it was called service discovery in earlier discussions, but this name is probably ill-suited). The idea is to allow resumable uploads to be transparently implemented by HTTP clients. This means that application developers just keep using the same API of their HTTP library as they have done in the past with traditional, non-resumable uploads. Once the HTTP library gets updated (e.g. because mobile OS or browsers start implementing resumable uploads), the HTTP library can transparently decide to use resumable uploads without explicit configuration by the application developer. Of course, in order to use resumable uploads, the HTTP library needs to know whether the server supports resumable uploads. If no support is detected, the HTTP library should use the traditional, non-resumable upload technique. We call this process feature detection.
+
+Ideally, the technique used for feature detection meets following **criteria** (there might not one approach which fits all requirements, so we have to prioritize them):
```
Comment (smatsson) [inline draft-tus-resumable-uploads-protocol.md]: Missing word. "there might not *be* one approach which fits a..."

File: draft-tus-resumable-uploads-protocol.md [hunk 2]
```diff
@@ -368,3 +368,29 @@ Specification: This document
...
+
+1. Avoid additional roundtrips by the client, if possible (i.e. an additional HTTP request by the client should be avoided).
+2. Be backwards compatible to HTTP/1.1 and existing network infrastructure: This means to avoid using new features in HTTP/2, or features which might require changes to existing network infrastructure (e.g. nginx or HTTP libraries)
+3. Conserve the user's privacy (i.e. the feature detection should not leak information other third-parties about which URLs have been connected to)
```
Comment (smatsson) [inline draft-tus-resumable-uploads-protocol.md]: "... leak information *to* other third-parties"?

**Review by smatsson — COMMENTED** [review]
Minor issues, otherwise good! 👍

**Comment by Acconut** [comment]
Thank you for the review, @smatsson!

**Pull request merged**